IBM/carbon-addons-iot-react
# fix(hierarchylist): changed scrollto to scrollintoview

Closes #947 

**Summary**

- Fixed scrollTo `defaultSelectedId` bug and horizontal scrollbar appearing when listItem's are longer than the list content
- Fixed multiple expanded parents for a single `defaultSelectedId`
- Fixed search function when there is only 1 column of values (no `secondaryValue`)

**Change List (commits, features, bugs, etc)**

- Updated search function to only search the `secondaryValue` if it is not nil
- Updated search for `defaultSelectedId` to only look for exact matches, not `includes()`
- Updated `scrollTo` function to use `scrollIntoView` as measurements were not working correctly
- Updated 1 `HierarchyList` story to have 1 very long player name, added 1 of the player names to another team (to test multiple id , for further testing of the above issues
- Added `overflow-x: hidden` to the list content container
- Added keys to a few stories to remove warnings

**Acceptance Test (how to verify the PR)**

- Check both `HierarchyList` stories and ensure that searching works for both
- Check the `defaultSelectedId` to ensure the element is scrolled to properly
- Check that the horizontal scroll bar does not appear on either story
